[#346] Document versioning policy (#352)
serokell/tezos-packaging: docs/versioning.md

<!--
   - SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
   -
   - SPDX-License-Identifier: LicenseRef-MIT-TQ
   -->

# Versioning

> `tezos-packaging` provides support for all Octez stable releases and RC candidates.

Reviewer: :thinking: `support` on it's own seems a bit odd IMO, it kinda sounds like this repository offers technical assistance on releases.
Also, the `C` in `RC` stands for candidate, so there is duplication there :slightly_smiling_face: 

What do you think of:
```suggestion
`tezos-packaging` follows all Octez releases, both stable and candidates.
```

However, in some cases we provide multiple releases within the same upstream version
because our packages provide additional functionality (e.g. systemd services in Ubuntu
and Fedora packages or brew formulae with launchd services) that may need to be updated

> within the single upstream release.

Reviewer: Minor suggestion, buy I would rephrase this to more have this order instead:
1. tezos-packaging releases can provide additional functionalities
2. give examples
3. (implied) these are tracked in addition top of the upstream version

P.s it sounds more correct to me to say "the same" rather than "the single" at the end there

In order to achieve that, our GitHub releases and packages use the following RPM-like versioning scheme:
`<upstream-version>-<release-number>`, e.g. `v11.0-1` for the first `tezos-packaging` release
within the `v11.0` upstream stable release, or `v11.0-rc2-2` for the second `tezos-packaging` release

> within the `v11.0-rc2` upstream release candidate.

Reviewer: from what i read only RPM versioning is more like:
> [name]-[version]-[release]

all of these 3 values apply to our versioning scheme too. So I suggest:
-  to mention that this scheme is applied to binaries too, which have a `name`
- I would use `version` more in the explanations to indicate the upstream releases

`<release-number>` is used to reflect the changes in additional packages functionality.

> ## Github releases

Reviewer: I think it would be better to always capitalize GitHub properly (like I just did :sweat_smile: )

> We provide at github releases for stable upstream releases and pre-releases for release candidates.

Reviewer: ```suggestion
We provide github releases for stable upstream releases and pre-releases for release candidates.
```

Github {pre-}releases contain static binaries and brew bottles compiled from the given
upstream source version.

## Ubuntu packages

> Ubuntu packages use slightly different versioning scheme, which is used to follow

Reviewer: ```suggestion
Ubuntu packages use a slightly different versioning scheme, which follows
```

<!--
   - SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
   -
   - SPDX-License-Identifier: LicenseRef-MIT-TQ
   -->

# Versioning

`tezos-packaging` follows all Octez releases, both stable and candidates.

Packages from our releases provide addinional functionality (e.g. systemd services
in Ubuntu and Fedora package or brew formulae with launchd services).

> This additional functionality may change within the same upstream version

Reviewer: ```suggestion
This additional functionality may change within the same upstream version.
```

> In order to track these additional functionality chages, our GitHub releases and packages use

Reviewer: Small suggestion:
```suggestion
In order to track this, our GitHub releases and packages use
```
for two reasons:
1. "changes" was misspelled
2. "additional functionality" is in both the previous two sentences as well as in one of the points just following, I think it's worth removing _some_ of this repetition :sweat_smile:
